File: Scripts/setup_user.py
import os
from supabase import create_client
from dotenv import load_dotenv

load_dotenv()

supabase_url = os.getenv("SUPABASE_URL") or os.getenv("VITE_SUPABASE_URL")
supabase_key = os.getenv("SUPABASE_SERVICE_ROLE_KEY")

if not supabase_url or not supabase_key:
    print("Missing Supabase credentials")
    exit(1)

supabase = create_client(supabase_url, supabase_key)

email = "timothynduva349@gmail.com"
password = "123456"

def setup_user():
    print(f"Setting up user: {email}")
    
    try:
        # Use auth admin to list users
        users_res = supabase.auth.admin.list_users()
        users = users_res.users
        
        existing_user = next((u for u in users if u.email == email), None)
        
        if existing_user:
            print(f"User exists (ID: {existing_user.id}). Updating...")
            res = supabase.auth.admin.update_user_by_id(
                existing_user.id,
                {
                    "password": password,
                    "user_metadata": {**existing_user.user_metadata, "beeyield_active": True, "role": "admin"},
                    "email_confirm": True
                }
            )
            print("User updated successfully!")
        else:
            print("User does not exist. Creating...")
            res = supabase.auth.admin.create_user({
                "email": email,
                "password": password,
                "user_metadata": {"beeyield_active": True, "role": "admin"},
                "email_confirm": True
            })
            print("User created successfully!")
            
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    setup_user()

File: Scripts/setup_user.js
import { createClient } from '@supabase/supabase-js';
// No dotenv needed, we'll use --env-file


const supabaseUrl = process.env.SUPABASE_URL || process.env.VITE_SUPABASE_URL;
const supabaseServiceKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseServiceKey) {
    console.error('Missing Supabase credentials');
    process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseServiceKey, {
    auth: {
        autoRefreshToken: false,
        persistSession: false
    }
});

const email = 'timothynduva349@gmail.com';
const password = '123456';

async function setupUser() {
    console.log(`Setting up user: ${email}`);

    // 1. Check if user exists
    const { data: { users }, error: listError } = await supabase.auth.admin.listUsers();

    if (listError) {
        console.error('Error listing users:', listError);
        return;
    }

    const existingUser = users.find(u => u.email === email);

    if (existingUser) {
        console.log(`User exists (ID: ${existingUser.id}). Updating password and metadata...`);
        const { data, error } = await supabase.auth.admin.updateUserById(
            existingUser.id,
            {
                password: password,
                user_metadata: { ...existingUser.user_metadata, beeyield_active: true, role: 'admin' },
                email_confirm: true
            }
        );

        if (error) {
            console.error('Error updating user:', error);
        } else {
            console.log('User updated successfully!');
        }
    } else {
        console.log('User does not exist. Creating new user...');
        const { data, error } = await supabase.auth.admin.createUser({
            email,
            password,
            user_metadata: { beeyield_active: true, role: 'admin' },
            email_confirm: true
        });

        if (error) {
            console.error('Error creating user:', error);
        } else {
            console.log('User created successfully!', data.user.id);
        }
    }
}

setupUser();
